### DISCUSSION TITLE
--reset-values flag

### DISCUSSION THREAD

**Discussion by kkoralsky** [helmfile/helmfile#593] [Ideas]
As of values control mode of helmfile, Im missing a way to enforce reset values from cli, even if helmfile.yaml manifest has:

 ```yaml
helmDefaults:
   reuseValues: true
 ```
I propose to add `--reset-values` cli flag to achieve that.

In my current environment, we have some apps that are deployed both with application CD pipeline (used by developers) and from CLI in IAC repo (used by devops). We need `reuseValues: true` in helmfile.yaml, so that *generally* devops wont overwrite anything running from CICD. But there are cases that we'd like to enforce "reset values control-mode" from CLI. Eg. when new values are added to the chart, `--reuse-values` helm flag does not work well.

I can prepare a PR for that. I just want to make sure we are on the same page that this would be useful in some cases and consistent (following a rule that cli is overriding declaration in yaml)

**Comment by yxxhero** [comment]
@kkoralsky reset-values is the default behavior. please give more details about your case.

**Comment by yxxhero** [comment]
@kkoralsky I think maybe you want to add reuseValues at the release level.

**Comment by kkoralsky** [comment]
If you have this `.helmDefaults.reuseValues` set to `true` in yaml manifest, there is no way to override this from command line. I know this is subtle, because you can always unset this value, and then invoke `helmfile upgrade/diff ...  `, but Im just feeling inconsistency here, that you cannot override this from cli alone.

reuseValues on release level would also be neat, but thats a different story: Im quite sure it would require more effort to implement and personally I dont have use case for this yet.

**Reply by yxxhero** [reply]
Pr is welcome
